tests/test_compare.py

```diff
@@ -103,3 +104,58 @@ def test_uncertaintychange(self):
                 if not accumulate_changes:
                     modelcp = cp(model)
         assert True, "Failure in uncertainty changing test"
+
+    def test_missing_uncertainties(self):
+        # Removes uncertainties from both models and attempts to use compare.
+
+        par_base1 = """
+            PSR J1234+5612
+            RAJ 14:34:01.00 
+            DECJ 56:14:00.00
+            F0 1 
+            PEPOCH 57000
+            DM 10 
+            DMEPOCH 57000
+            DM1 2
+            DMX     0.0
+            DMX_0001   1.0 
+            DMXR1_0001     58000.0
+            DMXR2_0001     58000.0
+            """
+
+        par_base2 = """
+            PSR J1234+5612
+            RAJ 14:34:01.00 
+            DECJ 56:14:00.00
+            F0 1  
+            PEPOCH 58000 
+            DM 10
+            DMEPOCH 57000
+            DM1 2 
+            DMX     0.0
+            DMX_0001   1.0 
+            DMXR1_0001     58000.0
+            DMXR2_0001     58000.0
+            """
+
+        model_1 = mod.get_model(io.StringIO(par_base1))
+        model_2 = mod.get_model(io.StringIO(par_base2))
+
+        for pn in model_1.params_ordered[1:]:
+            param1 = getattr(model_1, pn)
+            param2 = getattr(model_2, pn)
+            if (
+                param1 is None
+                or param2 is None
+                or param1.uncertainty is None
+                or param2.uncertainty is None
+            ):
+                continue
+            param1.frozen = False
+            param2.frozen = False
+            param1.uncertainty = 0.1 * param1.quantity
+            model_1.compare(model_2)
+            model_2.compare(model_1)
+            model_1 = mod.get_model(io.StringIO(par_base1))
```

Comment:
Why not just put these lines at the top of the loop instead of repeating them?

Reply:
not sure I understand. I want to reset the changes in each iteration of the loop so that I can only adjust one parameter at a time. but I need at least one model in advance for the loop, so i just made both there.

Reply:
It looks like you could just create the pair of models again each time through the loop? That way they're available the first time, and newly created every time after? And the code to create them is not duplicated?

Reply:
but i need the parameter list from the model for the loop.